### DISCUSSION TITLE
Handling "remainder" in a search term like query

### DISCUSSION THREAD

**Discussion by callumj** [alecthomas/participle#218] [General]
Hello,

Thank you so much for this project. I've read the examples and docs but I seem to be tripping up when I want to parse strings like this:

* `key:value another:value some remaining search text`
* `key:value another:value "quoted text"`

Where I want the non key values to be an array of strings (or in the case of the quoted string just one item).

I've been using this syntax:

```
type FullTextField struct {
	Fields  []*Field `@@*`
	Strings []*Value `@@*`
}

type Field struct {
	Key    string  `@Ident ":"`
	Values []Value `@@ ( "," @@ )*`
}

type Value struct {
	Str string `@Ident`
}
```

But I seem to be able to trip it up when a quote is used, as it doesn't seem to generate an invalid symbol.

**Comment by alecthomas** [comment]
I would probably structure this like so:

```go
type Search struct {
  Entry []*Entry `@@*`
}

type Entry struct {
  Field *Field `  @@`
  String *string `| @String`
  Value *string `| @Ident`
}
```

**Reply by alecthomas** [reply]
I just noticed that you will need a bit of lookahead to resolve this though because `Field` and `Value` both start with an `@Ident` and Participle will need to backtrack at that point.

**Comment by callumj** [comment]
That is exactly the pointer I needed, thanks @alecthomas!

**Reply by alecthomas** [reply]
You're welcome :)
